rust/abacus-base/src/inbox.rs

```diff
@@ -158,6 +162,14 @@ impl Inbox for InboxVariants {
             InboxVariants::Other(inbox) => inbox.message_status(leaf).await,
         }
     }
+
+    fn contract_address(&self) -> Address {
+        match self {
+            InboxVariants::Ethereum(inbox) => inbox.contract_address(),
+            InboxVariants::Mock(mock_inbox) => mock_inbox.contract_address(),
+            InboxVariants::Other(_) => unimplemented!(),
```

Comment:
You give me a hard time for panicing and then you write this 😬 

I would rather it be an option because the theory of this wrapper is that you are not supposed to know what the underlying is and then unwrapping the option is fine as long as we are willing to assert that we know what type it is, but if for some reason there is a section of code that _can't_ know then it would be wrong to panic inside like this. So yeah, generally, not very idiomatic but maybe tolerable given I am not sure we even want to keep this enum anyway.